source/joker_effects.c

```diff
@@ -1,31 +1,170 @@
 #include "game.h"
-#include "joker.h"
-#include "util.h"
 #include "hand_analysis.h"
+#include "joker.h"
 #include "list.h"
 #include "pool.h"
+#include "util.h"
+
 #include <stdlib.h>
 
+#define MISPRINT_MAX_MULT 23
 
-#define SCORE_ON_EVENT_ONLY_WITH_CARD(scored_card, restricted_event, checked_event, effect) \
-if (checked_event != restricted_event || scored_card == NULL) \
-{ \
-    return effect; \
-}
-#define SCORE_ON_EVENT_ONLY(restricted_event, checked_event, effect) \
-if (checked_event != restricted_event) \
-{ \
-    return effect; \
-}
+#define SCORE_ON_EVENT_ONLY_WITH_CARD(scored_card, restricted_event, checked_event, effect)                            \
+    if (checked_event != restricted_event || scored_card == NULL)                                                      \
+    {                                                                                                                  \
+        return effect;                                                                                                 \
+    }
+#define SCORE_ON_EVENT_ONLY(restricted_event, checked_event, effect)                                                   \
+    if (checked_event != restricted_event)                                                                             \
+    {                                                                                                                  \
+        return effect;                                                                                                 \
+    }
 
 // Joker Effect functions
+static JokerEffect joker_effect_noop(Joker* joker, Card* scored_card, enum JokerEvent joker_event);
```

Comment:
Not going to prefix the `s_` here. I don't see it adding anything to make this file more readable.